pyproject.toml

```diff
@@ -1,6 +1,53 @@
+[build-system]
+requires = ["setuptools>=80.0"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "mender-integration"
+version = "0.1.0"
+description = "Mender Integration Tests Suite"
+readme = "README.md"
+license = { text = "Apache-2.0" }
+
+requires-python = ">=3.12"
+
+classifiers = [
+    "Programming Language :: Python :: 3",
+    "License :: OSI Approved :: Apache Software License",
+    "Operating System :: OS Independent",
+]
+
+dependencies = [
+    "cryptography==45.0.7",
+    "docker==7.1.0",
+    "filelock==3.19.1",
+    "kubernetes==33.1.0",
+    "msgpack==1.1.1",
+    "py3dns==4.0.2",
+    "pymongo==4.14.1",
+    "pytest==8.4.1",
+    "pytest-html==4.1.1",
+    "pytest-metadata==3.1.1",
+    "pytest-timeout==2.4.0",
+    "pytest-xdist==3.8.0",
+    "redo==3.0.0",
+    "requests==2.32.5",
+    "urllib3==1.26.20",
+    "websockets==15.0.1",
+    "flaky==3.8.1",
+    "stripe==12.5.0",
+    "aiosmtpd==1.4.6",
+]
+
+[tool.setuptools]
+packages = ["tests", "testutils"]
+
+# include compose files so they can be used downstream
+[tool.setuptools.package-data]
+"*" = ["docker-compose*.yml", "other-components*.yml"]
 
 [tool.black]
-target-version = ['py38']
+target-version = ['py312']
```

Comment:
Careful with this, our CI uses a [hardcoded version for black](https://github.com/mendersoftware/mendertesting/blob/18e1e8c1e86ebabf63afaf816007dedee831ebbd/.gitlab-ci-check-python3-format.yml#L57).

This is something that I also really want to fix (in CI) but in order to not increase the scope of your work I would keep it in the existing version.